src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Source.cs

```diff
@@ -34,13 +35,18 @@ private static void FreeSymbolMap(MultiDictionary<string, ISymbol> symbolMap)
         public static Task<SymbolTreeInfo> GetInfoForSourceAssemblyAsync(
             Project project, Checksum checksum, bool loadOnly, CancellationToken cancellationToken)
         {
+            var solution = project.Solution;
+            var workspace = solution.Workspace;
+            var solutionKey = SolutionKey.ToSolutionKey(solution);
+
             var result = TryLoadOrCreateAsync(
-                project.Solution,
+                workspace,
+                solutionKey,
                 checksum,
                 loadOnly,
                 createAsync: () => CreateSourceSymbolTreeInfoAsync(project, checksum, cancellationToken),
                 keySuffix: "_Source_" + project.FilePath,
-                tryReadObject: reader => TryReadSymbolTreeInfo(reader, checksum, nodes => GetSpellCheckerAsync(project.Solution, checksum, project.FilePath, nodes)),
+                tryReadObject: reader => TryReadSymbolTreeInfo(reader, checksum, nodes => GetSpellCheckerAsync(workspace, solutionKey, checksum, project.FilePath, nodes)),
```

Comment:
⚠️ This is going to capture `Solution` via `project`. If that is not desired, read `project.FilePath` to a local variable before this call.

Reply:
that's a good idea.  it's not hugely important here (the task is kicked off immediately.  but it's still good for hygeiene.